happi/client.py

```diff
@@ -535,3 +535,34 @@ def find_config():
         # If found nothing
         raise EnvironmentError("No happi configuration file found. "
                                "Check HAPPI_CFG.")
+
+    def list_choices(self, field):
+        """
+        List all choices for a given field
+
+        Parameters
+        ----------
+        field : string
+            search field to list all possible choices for
+            i.e 'beamline', 'name', 'z', 'prefix', etc.
+
+        Raises
+        ------
+        SearchError
+            If no devices in the database have an entry for the given field
+
+        Returns
+        -------
+        field_choices : list
+            list of choices for a given field that are in the database
+        """
+        field_choices = []
+        all_devs = self.all_devices
+        for dev in all_devs:
+            try:  # Want to ignore error if 'dev' doesn't have 'field'
+                choice = getattr(dev, field)
+                if choice not in field_choices:
+                    field_choices.append(choice)
+            except AttributeError:
+                raise SearchError('No entries found with given field')
```

Comment:
I don't think you want to raise `SearchError` this will stop on the first device that doesn't have the field. Imagine if you are searching for a custom field that is only on some containers. In my mind `SearchError` should only be raised if there are no containers with that field in the database.

Reply:
This I'm still stuck on. I changed it so that I only raise `SearchError` if `field_choices` (which starts as an empty set) is still empty after iterating through all devices. This works as one can now do:
`client.choices_for_field('n_filters')`
and returns:
`{6, 10}`
But I still seem to be unable to show the choices for some fields, specifically `type` or `_id`. Still trying to sort out why this is and how to fix it.

Reply:
Those don't get set as attributes on the device, but will only show up if you look at `post()`. In general, users should not be searching these fields.